lib/cli-engine.js

```diff
@@ -302,6 +305,17 @@ function CLIEngine(options) {
      */
     this.options = assign(Object.create(defaultOptions), options || {});
 
+    /**
+     * cache used to not operate on files that haven't changed since last successful
+     * execution (e.g. file passed with no errors and no warnings
+     * @type {Object}
+     */
+    this.fileCache = fileEntryCache.create(path.resolve(this.options.cacheFile));
```

Comment:
You can disable the rule in this file of necessary. I'm more concerned with people directly touching an object that they shouldn't. If this should be private, feel free to use the underscore and disable the rule.